control/controlclient/direct.go

```diff
@@ -1074,82 +1078,42 @@ func (c *Direct) sendMapRequest(ctx context.Context, isStreaming bool, nu Netmap
 		}
 
 		metricMapResponseMap.Add(1)
-		if i > 0 {
+		if mapResIdx > 0 {
 			metricMapResponseMapDelta.Add(1)
-		}
-
-		if debug := resp.Debug; debug != nil {
-			if code := debug.Exit; code != nil {
-				c.logf("exiting process with status %v per controlplane", *code)
-				os.Exit(*code)
-			}
-			if debug.DisableLogTail {
-				logtail.Disable()
-				envknob.SetNoLogsNoSupport()
-			}
-			if sleep := time.Duration(debug.SleepSeconds * float64(time.Second)); sleep > 0 {
-				if err := sleepAsRequested(ctx, c.logf, timeoutReset, sleep, c.clock); err != nil {
-					return err
-				}
-			}
-		}
-
-		// For responses that mutate the self node, check for updated nodeAttrs.
-		if resp.Node != nil {
-			setControlKnobsFromNodeAttrs(resp.Node.Capabilities)
-		}
-
-		// Call Node.InitDisplayNames on any changed nodes.
-		initDisplayNames(cmpx.Or(resp.Node, sess.lastNode).View(), &resp)
-
-		nm := sess.netmapForResponse(&resp)
-		if nm.SelfNode == nil {
-			c.logf("MapResponse lacked node")
-			return errors.New("MapResponse lacked node")
-		}
-
-		if DevKnob.StripEndpoints() {
-			for _, p := range resp.Peers {
-				p.Endpoints = nil
+		} else {
+			if resp.Node == nil {
```

Comment:
if the first message you get is a keep alive, followed up by a map response which doesn't have the SelfNode this check won't fire

Reply:
I don't think that's legal. The very first MapResponse must be a full one.